File: CS/TreeListContextMenu/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using TreeListContextMenu.Models;

namespace TreeListContextMenu.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();
        }

        [ValidateInput(false)]
        public ActionResult TreeListPartial()
        {
            var model = NorthwindDataProvider.GetEmployees();
            return PartialView("_TreeListPartial", model);
        }

        public ActionResult TreeListMovePartial(int employeeID, int? reportsTo)
        {
            NorthwindDataProvider.MoveEmployee(employeeID, Convert.ToInt32(reportsTo));
            return TreeListPartial();
        }
    }
}

File: VB/TreeListContextMenu/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace TreeListContextMenu.Controllers
	Public Class HomeController
		Inherits Controller
		'
		' GET: /Home/

		Public Function Index() As ActionResult
			Return View()
		End Function


		Public Function TreeListPartial() As ActionResult
			Return PartialView(NorthwindDataProvider.GetEmployees())
		End Function


		Public Function TreeListMovePartial(ByVal employeeID As Integer, ByVal reportsTo? As Integer) As ActionResult
			NorthwindDataProvider.MoveEmployee(employeeID, Convert.ToInt32(reportsTo))
			Return PartialView("TreeListPartial", NorthwindDataProvider.GetEmployees())
		End Function


	End Class
End Namespace
